File: mlt.dtos/synology/SynoTask.cs
namespace mlt.dtos.synology;

public class SynoTask
{
    public string Destination { get; set; }

    public string Uri { get; set; }

    public string Title { get; set; }

    public string Id { get; set; }

    public DownloadStatus Status { get; set; }

    public long size { get; set; }

    public long size_downloaded { get; set; }
}

File: mlt.web/src/app/core/models/SynoTask.ts
import {DownloadStatus} from "../enums/DownloadStatus";

export class SynoTask {
  destination: string = '';
  uri: string = '';
  title: string = '';
  id: string = '';
  status: DownloadStatus = DownloadStatus.Waiting;
  size: number = 0;
  sizeDownloaded: number = 0;
  createdDateTime?: Date;
  completedDateTime?: Date;

  constructor(init?: Partial<SynoTask>) {
    Object.assign(this, init);
  }
}
